components/dialog/index.css

```diff
@@ -34,337 +34,321 @@ governing permissions and limitations under the License.
+    margin-bottom: 0;
   }
 }
 
-.spectrum-Dialog-title {
-  flex: 1 1 auto;
+.spectrum-Dialog-buttonGroup {
+  grid-area: buttonGroup;
+  /* this padding isn't built into the grid because it disappears with this buttonGroup */
+  padding-block-start: var(--spectrum-global-dimension-static-size-500);
+  display: flex;
+  justify-content: flex-end;
+  /* this padding should be safe as button group is always end aligned */
+  padding-inline-start: var(--spectrum-dialog-gap-size);
 
-  margin: 0;
+  &.spectrum-Dialog-buttonGroup--noFooter {
+    grid-area: footer-start / footer-start / buttonGroup-end / buttonGroup-end;
+  }
+}
 
-  font-size: var(--spectrum-dialog-title-text-size);
-  font-weight: var(--spectrum-dialog-title-text-font-weight);
-  line-height: var(--spectrum-dialog-title-text-line-height);
+.spectrum-Dialog.spectrum-Dialog--dismissable .spectrum-Dialog-grid {
+  grid-template-columns: var(--spectrum-dialog-padding) auto 1fr auto minmax(0, auto) minmax(0, var(--spectrum-dialog-close-button-size)) var(--spectrum-dialog-padding);
+  grid-template-rows: auto var(--spectrum-dialog-padding) auto auto 1fr auto var(--spectrum-dialog-padding);
+  grid-template-areas:
+    "hero hero    hero    hero        hero        hero        hero"
+    ".    .       .       .           .           closeButton closeButton"
+    ".    heading header  header      typeIcon    closeButton closeButton"
+    ".    divider divider divider     divider     divider     ."
+    ".    content content content     content     content     ."
+    ".    footer  footer  buttonGroup buttonGroup buttonGroup ."
+    ".    .       .       .           .           .           .";
+
+  .spectrum-Dialog-buttonGroup {
+    display: none;
+  }
+  .spectrum-Dialog-footer {
+    grid-area: footer / footer/ buttonGroup / buttonGroup;
+  }
+}
 
-  outline: none; /* Hide focus outline */
+.spectrum-Dialog-closeButton {
+  grid-area: closeButton;
+  /* align and justify so it doesn't do the default 'stretch' and end up with forced height/width */
+  align-self: start;
+  justify-self: end;
+
+  margin-inline-end: var(--spectrum-dialog-close-button-padding);
+  margin-block-start: var(--spectrum-dialog-close-button-padding);
 }
 
 /* Alert Dialog is a specific type of Dialog */
-.spectrum-Dialog--alert {
-  inline-size: var(--spectrum-dialog-alert-width);
+.spectrum-Dialog--error {
+  width: var(--spectrum-dialog-alert-width);
 }
 
-/** @unofficial */
 .spectrum-Dialog--fullscreen {
-  top: var(--spectrum-dialog-fullscreen-margin);
-  right: var(--spectrum-dialog-fullscreen-margin);
-  bottom: var(--spectrum-dialog-fullscreen-margin);
-  left: var(--spectrum-dialog-fullscreen-margin);
-
-  transform: translate(0, var(--spectrum-dialog-entry-animation-distance));
-
-  &.is-open {
-    transform: translate(0, 0);
-  }
+  width: 100%;
+  height: 100%;
 }
+.spectrum-Dialog--fullscreenTakeover {
+  width: 100%;
+  height: 100%;
 
+  border-radius: 0;
+}
+/** @unofficial */
```

Comment:
why is this @unofficial?

Reply:
This comes from @snowystinger and I think it means that the fullscreen and fullscreen takeover are in contributions.